src/editors/hooks.test.jsx

```diff
@@ -0,0 +1,95 @@
+import { useEffect, updateState } from 'react';
+import * as module from './hooks';
+
+jest.mock('react', () => {
+  const updateStateMock = jest.fn();
+  return {
+    updateState: updateStateMock,
+    useState: jest.fn(val => ([{ state: val }, (newVal) => updateStateMock({ val, newVal })])),
+    useRef: jest.fn(val => ({ current: val })),
+    useEffect: jest.fn(),
+    useCallback: (cb, prereqs) => ({ cb, prereqs }),
+  };
+});
+describe('hooks', () => {
+  const locationTemp = window.location;
+  beforeAll(() => {
+    delete window.location;
+    window.location = {
+      assign: jest.fn(),
+    };
+  });
+  afterAll(() => {
+    window.location = locationTemp;
+  });
+  describe('initializeApp', () => {
+    const fakedata = { some: 'data' };
+    const mockIntialize = jest.fn(val => (val));
+    test('calls provided function with provided data as args when useEffect is called', () => {
+      module.initializeApp({ initialize: mockIntialize, data: fakedata });
+      expect(mockIntialize).not.toHaveBeenCalledWith(fakedata);
+      const [cb, prereqs] = useEffect.mock.calls[0];
+      expect(prereqs).toStrictEqual([]);
+      cb();
+      expect(mockIntialize).toHaveBeenCalledWith(fakedata);
+    });
+  });
+  describe('prepareEditorRef', () => {
+    let output;
+    beforeEach(() => {
+      output = module.prepareEditorRef();
+    });
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+    test('sets refReady to false by default, ref is null', () => {
```

Comment:
should have a test that specifically calls out the full set of outputs
"returns an object containing editorRef, refReady, and setEditorRef values"

below that, you can then have the tests of 
* "refReady defaults to false, editorRef defaults to null"
* "when useEffect triggers, refReady is set to true"
* "calling setEditorRef updates the editorRef value"